### PR TITLE
Add java solution for coin combinations II

### PR DESCRIPTION
**Pull request by nathangong**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as
done._

- [x] I have tested my code
- [x] I have followed the code style guidelines mentioned
      [here](https://usaco.guide/general/adding-solution/#code-conventions)
- [x] I have properly formatted the files according to the steps mentioned
      [here](https://usaco.guide/general/adding-solution#steps)

### CODE REVIEW

File: solutions/cses-1636.mdx [hunk 1]
```diff
@@ -77,4 +90,40 @@ int main(){
...
 
+<JavaSection>
+
+```java
+import java.util.*;
+
+public class CoinCombinations2 {
+	static final int MOD = 1000000007;
+
+    public static void main(String[] args) {
```
Comment (SansPapyrus683) [inline solutions/cses-1636.mdx]: tabs pwease uwu

File: solutions/cses-1636.mdx [hunk 2]
```diff
@@ -77,4 +90,40 @@ int main(){
...
+    	Scanner sc = new Scanner(System.in);
+    	int n = sc.nextInt();
+    	int x = sc.nextInt();
+
+    	int[] coins = new int[n];
+    	for (int i = 0; i < n; i++) {
+    		coins[i] = sc.nextInt();
+    	}
+
+    	int[] dp = new int[x + 1];
```
Comment (SansPapyrus683) [inline solutions/cses-1636.mdx]: short comment explaining what this holds

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/cses-1636.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/cses-1636.mdx]: wait nvm you just fixed it

**Review by SansPapyrus683 — APPROVED** [review]

File: solutions/cses-1636.mdx [hunk 3]
```diff
@@ -77,4 +90,38 @@ int main(){
 
 </CPPSection>
 
+<JavaSection>
+
+```java
+import java.util.*;
+
+public class CoinCombinations2 {
+	static final int MOD = 1000000007;
```
Comment (SansPapyrus683) [inline solutions/cses-1636.mdx:99]: (optional)
```suggestion
	static final int MOD = (int) 1e9 + 7;
```

**Review by dongliu0426 — APPROVED** [review]

**Pull request merged**